tls/s2n_ktls.c

```diff
@@ -15,11 +15,147 @@
 
 #include "tls/s2n_ktls.h"
 
+#include <sys/socket.h>
+
+#include "utils/s2n_socket.h"
+
 bool s2n_ktls_is_supported_on_platform()
 {
-#if defined(__linux__)
+#if S2N_KTLS_SUPPORTED
     return true;
 #else
     return false;
 #endif
 }
+
+S2N_RESULT s2n_ktls_validate(struct s2n_connection *conn, s2n_ktls_mode ktls_mode)
```

Comment:
Probably worth marking any "private" method as "static" to make the scope clear (same with your tests)